src/ibex_bluesky_core/devices/polarisingdae/__init__.py

```diff
@@ -0,0 +1,252 @@
+"""An interface to the DAE for bluesky, suited for polarisation."""
+
+import logging
+from typing import Generic, TypeAlias
+
+import scipp as sc
+from bluesky.protocols import Movable, Triggerable
+from ophyd_async.core import (
+    AsyncStageable,
+    AsyncStatus,
+    Reference,
+)
+from typing_extensions import TypeVar
+
+from ibex_bluesky_core.devices.dae import Dae
+from ibex_bluesky_core.devices.polarisingdae._reducers import (
+    MultiWavelengthBandNormalizer,
+    PolarisationReducer,
+)
+from ibex_bluesky_core.devices.simpledae import (
+    Controller,
+    GoodFramesWaiter,
+    PeriodGoodFramesWaiter,
+    PeriodPerPointController,
+    Reducer,
+    RunPerPointController,
+    Waiter,
+    wavelength_bounded_spectra,
+)
+from ibex_bluesky_core.utils import get_pv_prefix
+
+logger = logging.getLogger(__name__)
+
+__all__ = [
+    "DualRunDae",
+    "MultiWavelengthBandNormalizer",
+    "PolarisationReducer",
+    "polarising_dae",
+]
+
+TController_co = TypeVar("TController_co", bound="Controller", default=Controller, covariant=True)
+TWaiter_co = TypeVar("TWaiter_co", bound="Waiter", default=Waiter, covariant=True)
+TPReducer_co = TypeVar(
+    "TPReducer_co",
+    bound="Reducer",
+    default=Reducer,
+    covariant=True,
+)
+TMWBReducer_co = TypeVar(
+    "TMWBReducer_co",
+    bound="Reducer",
+    default=Reducer,
+    covariant=True,
+)
+
+
+class DualRunDae(
+    Dae,
+    Triggerable,
+    AsyncStageable,
+    Generic[TController_co, TWaiter_co, TPReducer_co, TMWBReducer_co],
+):
+    """DAE with strategies for data collection, waiting, and reduction, suited for polarisation.
+
+    This class is a more complex version of SimpleDae. It requires a flipper device to be provided
+    and will perform two runs, changing the flipper device at the start and inbetween runs.
+    """
+
+    def __init__(  # noqa: PLR0913
+        self,
+        *,
+        prefix: str,
+        name: str = "DAE",
+        controller: TController_co,
+        waiter: TWaiter_co,
+        reducer_final: TPReducer_co,
+        reducer_up: TMWBReducer_co,
+        reducer_down: TMWBReducer_co,
+        flipper: Movable[float],
+        flipper_states: list[float],
```

Comment:
I wonder if this could actually be generic (rather than just flipper), I think it'd work with any movable?